src/main/java/com/hermesworld/ais/galapagos/topics/service/impl/TopicServiceImpl.java

```diff
@@ -145,6 +145,61 @@ public void init(KafkaCluster cluster) {
         });
     }
 
+    @Override
+    public CompletableFuture<Void> addTopicProducers(String environmentId, String topicName, List<String> producers) {
+
+        if (getTopic(environmentId, topicName).isEmpty()) {
+            return CompletableFuture
+                    .failedFuture(new IllegalStateException("Topic with name " + topicName + " does not exist!"));
+        }
+        TopicMetadata topic = getTopic(environmentId, topicName).get();
```

Comment:
Don't write all that code. Use `doWithClusterAndTopic` helper method. (But the COMMANDS check below must make it into your passed lambda expression :-))